File: src/main/java/net/arvandor/numinoustreasury/item/action/OpenInventory.java
package net.arvandor.numinoustreasury.item.action;

import net.arvandor.numinoustreasury.item.NuminousItemInventoryHolder;
import net.arvandor.numinoustreasury.item.NuminousItemStack;
import org.bukkit.event.player.PlayerInteractEvent;
import org.bukkit.inventory.ItemStack;

import java.util.Map;

public final class OpenInventory implements NuminousOnInteractAir, NuminousOnInteractBlock {

    @Override
    public void onInteractAir(PlayerInteractEvent event) {
        openInventory(event);
    }

    @Override
    public void onInteractBlock(PlayerInteractEvent event) {
        openInventory(event);
    }

    private static void openInventory(PlayerInteractEvent event) {
        ItemStack bukkitItem = event.getItem();
        NuminousItemStack numinousItem = NuminousItemStack.fromItemStack(bukkitItem);
        if (numinousItem == null) return;
        NuminousItemInventoryHolder holder = new NuminousItemInventoryHolder(numinousItem, bukkitItem);
        event.getPlayer().openInventory(holder.getInventory());
    }

    @Override
    public Map<String, Object> serialize() {
        return Map.of();
    }

    public static OpenInventory deserialize(Map<String, Object> args) {
        return new OpenInventory();
    }
}

File: src/main/kotlin/net/arvandor/numinoustreasury/item/action/OpenInventory.kt
package net.arvandor.numinoustreasury.item.action

import net.arvandor.numinoustreasury.item.NuminousItemInventoryHolder
import net.arvandor.numinoustreasury.item.NuminousItemStack
import org.bukkit.event.player.PlayerInteractEvent

class OpenInventory : NuminousOnInteractAir, NuminousOnInteractBlock {
    override fun onInteractAir(event: PlayerInteractEvent) {
        openInventory(event)
    }

    override fun onInteractBlock(event: PlayerInteractEvent) {
        openInventory(event)
    }

    override fun serialize(): Map<String, Any> {
        return mapOf()
    }

    companion object {
        private fun openInventory(event: PlayerInteractEvent) {
            val bukkitItem = event.item
            val numinousItem: NuminousItemStack = NuminousItemStack.Companion.fromItemStack(bukkitItem) ?: return
            val holder = NuminousItemInventoryHolder(numinousItem, bukkitItem)
            event.player.openInventory(holder.inventory)
        }

        @JvmStatic
        fun deserialize(serialized: Map<String?, Any?>?): OpenInventory {
            return OpenInventory()
        }
    }
}
